mlflow/store/artifact/databricks_artifact_repo.py

```diff
@@ -202,25 +242,33 @@ def _azure_upload_file(self, credentials, local_file, artifact_path):
         try:
             headers = self._extract_headers_from_credentials(credentials.headers)
             uploading_block_list = []
-            for chunk in yield_file_in_chunks(local_file, _AZURE_MAX_BLOCK_CHUNK_SIZE):
-                # Base64-encode a UUID, producing a UTF8-encoded bytestring. Then, decode
-                # the bytestring for compliance with Azure Blob Storage API requests
-                block_id = base64.b64encode(uuid.uuid4().hex.encode()).decode("utf-8")
-                try:
-                    put_block(credentials.signed_uri, block_id, chunk, headers=headers)
-                except requests.HTTPError as e:
-                    if e.response.status_code in [401, 403]:
-                        _logger.info(
-                            "Failed to authorize request, possibly due to credential expiration."
-                            " Refreshing credentials and trying again..."
+            futures = {}
+            with self.thread_pool as executor:
+                for chunk_index, chunk in enumerate(
+                    yield_file_in_chunks(local_file, _AZURE_MAX_BLOCK_CHUNK_SIZE)
+                ):
+                    upload_future = executor.submit(
+                        self._azure_upload_chunk,
+                        credentials=credentials,
+                        chunk=chunk,
+                        local_file=local_file,
+                        artifact_path=artifact_path,
+                    )
+                    futures[upload_future] = chunk_index
+
+                    # Avoid loading too many chunks into memory
+                    if len(futures) < 5:
```

Comment:
Can we use a smaller chunk size so that we can execute more futures at once? @BenWilson2 mentioned that 8-12MB might be good. Maybe we can start with 10MB and benchmark the improvement?

Reply:
Sounds good :)

Reply:
Decreased the chunk size to 10 MB